src/gcode-parser.ts

```diff
@@ -128,22 +174,23 @@ export class Parser {
   parseGCode(input: string | string[]): {
     layers: Layer[];
     metadata: Metadata;
+    commands: GCodeCommand[];
   } {
     const lines = Array.isArray(input) ? input : input.split('\n');
 
     this.lines = this.lines.concat(lines);
 
-    const commands = this.lines2commands(lines);
+    this.commands = this.lines2commands(lines);
 
-    this.groupIntoLayers(commands);
+    // this.groupIntoLayers(this.commands);
```

Comment:
I am thinking about letting the interpreter group into layers, but to not have rendering logic looping on layers by default. This would allow more flexibility for rendering non-planar gcodes.

Reply:
For min/max layer I plan to use clipping planes. This means we can change the min/max without having to rerender the 3d object.

So layer grouping can be removed from the parser. 
For now, the layer height is a constructor param but eventually we want the parser to read this from the gcode.

Reply:
I don't think we should remove layers completely. We'll have interesting challenges when comes the time to support adaptive layers properly! But right now, we're able to completely decouple rendering from layers.